SMatrixADT.c

```diff
@@ -339,7 +339,7 @@ int SMTX_multiply(SMatrixType A[],SMatrixType B[],SMatrixType C[])
             while(col_B == B->R[j].row && j <= totb)
                 j++;
         }
-        while(row_A = A->R[i].row && i <= tota)  //
+        while(row_A == A->R[i].row && i <= tota)  //
```

Comment:
"while(row_A = A->R[i].row && i <= tota)" should use "==" for comparison, not "=" for assignment